### PR TITLE
Bridge in rust

### PR DESCRIPTION
**Pull request by bakulf**
# Intro
This is a new implementation of mozillavpnnp in rust.

# Why
Running the unit-tests for the mozillavpnnp component on windows I found several bugs:
1. the STDIN reading/writing was totally wrong. We were using stdin to synchronize the socket reading; some messages were not received; etc..
2. the TCP handling was wrong too! The connection was sync (blocking all the rest); big windows timeouts

I tried to cleanup the mess and the code was a collection of `#ifdef MVPN_WINDOWS...`. Better to rewrite it.

# Next step
1. a proper review by rust experts
2. integration with the compilation scripts

### CODE REVIEW

File: extension/bridge/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+                                    }
+                                }
+                            }
+                            _ => unreachable!(),
+                        }
+                    }
+                }
+            }
+            _ => {
+                eprintln!("No connection to the VPN client");
```
Comment (oskirby) [inline extension/bridge/src/main.rs:121]: Wouldn't this get kind of spammy every time the user loads their browser without the VPN client running?

File: extension/bridge/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+[package]
+name = "mozillavpnnp"
+version = "0.1.0"
```
Comment (oskirby) [inline extension/bridge/Cargo.toml:3]: This is probably fine for testing, but we should find some way for this to track the version of the main VPN client app.

**Review by oskirby — COMMENTED** [review]
Very cool. I think this is a great improvement. However, we need to update the README and build instructions to mention that at Rust toolchain is now required when building the web extension.

Likewise, the Linux packaging needs updates too to mention Rust as a Build-Depends.

**Comment by bakulf** [comment]
This is an intermediate step. I want to enable platforms one at a time in separate PR. Would it work for you?

**Review by oskirby — APPROVED** [review]

**Pull request merged**